src/Internal/Service/PaymentProcessingService.php

```diff
@@ -8,19 +8,129 @@
+use WC_Payment_Gateway_WCPay;
+use WC_Payments_Subscription_Service;
+use WCPay\Internal\Payment\Flag;
+use WCPay\Internal\Payment\PaymentRequest;
+use WCPay\Internal\Payment\Response\Failure;
+use WCPay\Internal\Payment\Storage;
+use WCPay\Internal\Payment\Response\ResponseInterface;
+use WCPay\Internal\Proxy\LegacyProxy;
+use WCPay\WooPay\WooPay_Utilities;
 
 /**
  * Payment Processing Service.
  */
 class PaymentProcessingService {
 	/**
-	 * Process payment.
+	 * Storage for payments.
+	 *
+	 * @var Storage
+	 */
+	private $storage;
+
+	/**
+	 * Legacy proxy.
+	 *
+	 * @var LegacyProxy
+	 */
+	private $legacy_proxy;
+
+	/**
+	 * Subscriptions service.
+	 *
+	 * @var WC_Payments_Subscription_Service
+	 */
+	private $subscriptions_service;
+
+	/**
+	 * WooPay utils.
+	 *
+	 * @var WooPay_Utilities
+	 */
+	private $woopay_utilities;
+
+	/**
+	 * Class constructor.
 	 *
-	 * @param  int $order_id Order ID provided by WooCommerce core.
+	 * @param Storage                          $storage               Payment storage.
+	 * @param LegacyProxy                      $legacy_proxy          Legacy proxy.
+	 * @param WC_Payments_Subscription_Service $subscriptions_service Subscriptions service.
+	 */
+	public function __construct(
+		Storage $storage,
+		LegacyProxy $legacy_proxy,
+		WC_Payments_Subscription_Service $subscriptions_service,
+		WooPay_Utilities $woopay_utilities
+	) {
+		$this->storage               = $storage;
+		$this->legacy_proxy          = $legacy_proxy;
+		$this->subscriptions_service = $subscriptions_service;
+		$this->woopay_utilities      = $woopay_utilities;
+	}
+
+	/**
+	 * Process payment.
 	 *
-	 * @throws Exception
+	 * @param int  $order_id       Order ID provided by WooCommerce core.
+	 * @param bool $manual_capture Whether the payment should be captured manually. Defaults to false.
+	 * @return ResponseInterface   The response from processing the payment.
 	 */
-	public function process_payment( int $order_id ) {
-		throw new Exception( 'Re-engineering payment process is in-progress. Sit tight, and wait more!' );
+	public function process_payment( int $order_id, bool $manual_capture = false ) {
+		$order = wc_get_order( $order_id ); // ToDo: This function should not be called directly, but through a service!
+
+		if ( ! $order instanceof WC_Order ) {
+			return new Failure( __( 'Processing payment failed: Order could not be loaded.', 'woocommerce-payments' ) );
+		}
+
+		try {
+			// Warning: Dev mode only! Will be deleted soon.
+			$this->storage->enable();
+			$this->storage->cleanup_order_payment( $order );
+
+			$post_data = [
+				'payment_method'                             => WC_Payment_Gateway_WCPay::GATEWAY_ID,
+				'wcpay-payment-method'                       => 'pm_XYZ',
+				'wc-woocommerce_payments-new-payment-method' => true,
+			];
+
+			// Initialize the request and payment.
+			$request = new PaymentRequest( $this->legacy_proxy, $post_data );
+			$payment = $this->storage->get_order_payment( $order );
```

Comment:
How about considering the current state as "payment" throughout methods like this one (`process_payment`)? It makes sense to me that externally, the process should be concerned about payment vs. state as little as possible?

Reply:
> But the classes themselves should mention states, see my other comment.

I'm still to get there, but they do. 😕